app/models/loan_repayment.rb

```diff
@@ -66,7 +66,7 @@ class LoanRepayment < Ekylibre::Record::Base
   }
 
   before_validation do
-    self.amount = base_amount + insurance_amount + interest_amount
+    self.amount = base_amount ||= 0 + insurance_amount ||= 0 + interest_amount ||= 0
```

Comment:
I didn't think that default assignment could be done inline. I don't know if this is causing the failure but the tests fail.
Maybe you can try to add them on separate lines:
```ruby
base_amount ||= 0
insurance_amount ||= 0
...
```